### PR TITLE
Github action for tests

### PR DESCRIPTION
**Pull request by aristidesstaffieri**
[Ticket](https://stellarorg.atlassian.net/jira/software/c/projects/WAL/boards/36?modal=detail&selectedIssue=WAL-743)

Adds a mock local store.
Fixes previously failing or flakey tests.
Adds an action to run test on PRs.
I've opened [this PR in terraform](https://github.com/stellar/terraform/pull/1271) to block merge on a failing status check from our CI tests.

### CODE REVIEW

File: extension/src/background/messageListener/__tests__/popupMessageListener.test.js [hunk 1]
```diff
@@ -9,11 +9,24 @@ import {
...
+async function getKeyId(pubKey, privKey) {
+  const str = pubKey + privKey;
+  const enc = new TextEncoder();
+  const hash = await crypto.subtle.digest("SHA-1", enc.encode(str));
+  return Array.from(new Uint8Array(hash))
+    .map((v) => v.toString(16).padStart(2, "0"))
+    .join("");
+}
+
+describe.skip("regular account flow", () => {
```
Comment (aristidesstaffieri) [inline extension/src/background/messageListener/__tests__/popupMessageListener.test.js:26]: @piyalbasu I'm hoping you can take a look at this test suite with me, it's the last one I haven't been able to resolve. Basically the KeyManger errors with `Some keys were already stored in the keystore` but I'm not sure of why. We clear local storage in between tests and if I log out the keyManager keys just before it fails I see an empty list 🤔

File: extension/src/background/messageListener/popupMessageListener.ts [hunk 2]
```diff
@@ -233,6 +233,7 @@ export const popupMessageListener = (request: Request) => {
     let keyStore = { id: "" };
 
     try {
+      // const key = await keyManager.loadAllKeyIds();
```
Comment (piyalbasu) [inline extension/src/background/messageListener/popupMessageListener.ts:236]: Watch out for merging this. Let's just comment out any flakey tests and we can fix those in a follow up PR

File: extension/src/background/messageListener/__tests__/popupMessageListener.test.js (hunk 1 above)
Comment (piyalbasu) [reply extension/src/background/messageListener/__tests__/popupMessageListener.test.js:26]: Ah yeah, we can look at that. let's create a new ticket for dealing with any broken tests. We can comment out failing tests just to get this merged

File: extension/src/background/messageListener/popupMessageListener.ts (hunk 2 above)
Comment (Mateusz9611) [reply extension/src/background/messageListener/popupMessageListener.ts:236]: results.length

**Review by Mateusz9611 — APPROVED** [review]
94951a70c4198e4b9ef10ce0fc5cd195c6c5a2be

**Pull request closed without merging**